sklearn/cluster/_kmeans.py

```diff
@@ -1169,48 +1176,42 @@ def _more_tags(self):
         }
 
 
-def _mini_batch_step(X, sample_weight, x_squared_norms, centers, weight_sums,
-                     old_center_buffer, compute_squared_diff,
-                     distances, random_reassign=False,
-                     random_state=None, reassignment_ratio=.01,
-                     verbose=False):
+def _mini_batch_step(X, x_squared_norms, sample_weight, centers, centers_new,
+                     weight_sums, random_state, random_reassign=False,
+                     reassignment_ratio=0.01, verbose=False, n_threads=1):
     """Incremental update of the centers for the Minibatch K-Means algorithm.
 
     Parameters
     ----------
 
-    X : ndarray of shape (n_samples, n_features)
-        The original data array.
-
-    sample_weight : array-like of shape (n_samples,)
-        The weights for each observation in X.
+    X : {ndarray, sparse matrix} of shape (n_samples, n_features)
+        The original data array. In sparse, must be in CSR format.
 
     x_squared_norms : ndarray of shape (n_samples,)
         Squared euclidean norm of each data point.
 
-    centers : ndarray of shape (k, n_features)
-        The cluster centers. This array is MODIFIED IN PLACE
+    sample_weight : ndarray of shape (n_samples,)
+        The weights for each observation in X.
 
-    counts : ndarray of shape (k,)
-         The vector in which we keep track of the numbers of elements in a
-         cluster. This array is MODIFIED IN PLACE
+    centers : ndarray of shape (n_clusters, n_features)
+        The cluster centers before the current iteration
 
-    distances : ndarray of shape (n_samples,), dtype=float, default=None
-        If not None, should be a pre-allocated array that will be used to store
-        the distances of each sample to its closest center.
-        May not be None when random_reassign is True.
+    centers_new : ndarray of shape (n_clusters, n_features)
+        The cluster centers after the current iteration. Modified in-place.
 
-    random_state : int, RandomState instance, default=None
-        Determines random number generation for centroid initialization and to
-        pick new clusters amongst observations with uniform probability. Use
-        an int to make the randomness deterministic.
+    weight_sums : ndarray of shape (n_clusters,)
+        The vector in which we keep track of the numbers of points in a
+        cluster. This array is modified in place.
+
+    random_state : RandomState instance
+        Determines random number generation for low count centers reassignment.
         See :term:`Glossary <random_state>`.
 
-    random_reassign : bool, default=None
+    random_reassign : boolean, default=False
         If True, centers with very low counts are randomly reassigned
         to observations.
 
-    reassignment_ratio : float, default=None
+    reassignment_ratio : float, default=0.01
```

Comment:
Shall we change the value of the `reassignment_ratio` based on your experiments or is this good enough?

Reply:
I'm not sure this experiment is enough to change it. This results are averages over several runs. Increasing the reassignement ratio seems to give better chances to find a better minima but it can also find a worse minima. A low value seems safer and more stable to me.